uwblueprint/paramedics-react
# Edit and delete location pins

[Notion ticket](https://www.notion.so/uwblueprintexecs/3-Edit-and-delete-location-pins-233db29d406e4fc79aed2b6b660d80f6)

[Edit pin Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1174%3A21142)
[Delete pin Figma link](https://www.figma.com/file/IpOPyv79gBwEohvOJvPV6f/%F0%9F%9A%91-Designs-for-Development-S20-F20?node-id=1174%3A21284)

## Changes
- Added edit pin flow
    - Click on a pin -> click edit -> change it up and confirm -> pin has been edited!
    - Info window doesn't close if you click edit (I thought this would be useful for the paramedics since they can continue to refer to the pin even though it has been edited
- Added delete pin flow
    - Canceling pin deletion doesn't close info window (again, same reason as above)
- Changed a bit of confirm modal as it was calling `handleCancel`, which was messing up the expected behaviour of canceling a deletion
## Screenshots
About to edit a pin
<img width="933" alt="Screen Shot 2020-11-24 at 6 10 04 PM" src="https://user-images.githubusercontent.com/21224282/100169260-47c95300-2e80-11eb-9d64-9417304de37d.png">
After edit completion:
<img width="948" alt="Screen Shot 2020-11-24 at 6 08 13 PM" src="https://user-images.githubusercontent.com/21224282/100169165-0638a800-2e80-11eb-877d-0ba955b1ad11.png">

Deletion (not completely updated in screenshot, but information window will be open at this time according to Figma:
<img width="941" alt="Screen Shot 2020-11-24 at 6 09 37 PM" src="https://user-images.githubusercontent.com/21224282/100169242-384a0a00-2e80-11eb-8a04-854af922f693.png">

## Testing
- Test the happy paths:
    - Edit a pin and make sure the changes are properly reflected (change both the name of the pin as well as the location)
    - Delete a pin and make sure the changes are properly reflected on both the front end and the backend
- Make sure cancellation functionality is working: for both edit and delete, canceling the flow should leave the info window open
- Create three pins and perform the following:
    - 3 consecutive deletes
    - 3 consecutive edits
    - Edit a pin -> click add (fields should be cleared) -> completion/cancellation should clear the info window (you should not see it)
    - Delete -> edit -> add
    - Any other combinations

**Note: If you have any suggestions/improvements on my state management, I will gladly take it!! It seems messy to me**

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
